### PR TITLE
CoreDNS Operator: Prefer to get IP from kube-dns service

### PR DESCRIPTION
**Pull request by justinsb**
If the kube-dns service already exists, we should use that ClusterIP.

### CODE REVIEW

**Comment by atoato88** [comment]
Prow job [failed](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-addons/104/pull-cluster-addons-unit-test/1435248496734113792) because that golden file test is failed.

But The test will pass if #103 is merged, because #103 updates [kubebuilder-declarative-pattern](https://github.com/kubernetes-sigs/cluster-addons/pull/103/files#diff-34e449582f888a45b7937efe90bb6cd8447391cd87cd138e6fcf055561ba4b74R13) module, right?
So this PR depends on #103 I think.

File: coredns/controllers/util.go [hunk 1]
```diff
@@ -70,14 +78,28 @@ func getCoreDNSDeployment(ctx context.Context, c client.Client) (*appsv1.Deploym
...
-		return "", err
+	kubeDNSService, err := getKubeDNSService(ctx, c)
+	if err != nil {
+		if !apierrors.IsNotFound(err) {
+			return "", err
+		}
+	} else {
+		ip := net.ParseIP(kubeDNSService.Spec.ClusterIP)
+		if ip == nil {
+			return "", fmt.Errorf("cannot parse kube-dns ClusterIP %q", kubeDNSService.Spec.ClusterIP)
```
Comment (stealthybox) [inline coredns/controllers/util.go:89]: I considered suggesting we fall-through here to the heuristic behavior, but if the kube-dns Service object already exists without a ClusterIP, it's probably best to fail now and try later like you have it.

**Comment by atoato88** [comment]
Is this PR still alive?

**Pull request closed without merging**